pynest/examples/brette_et_al_2007/cuba_stdp.py

```diff
@@ -0,0 +1,262 @@
+I_neurons = nest.Create(model, NI)
+
+print("Creating stimulus generators...")
+E_stimulus = nest.Create("poisson_generator")
+E_stimulus.set(stimulus_params)
+
+I_stimulus = nest.Create("poisson_generator")
+I_stimulus.set(stimulus_params)
+
+print("Creating spike recorders...")
+E_recorder = nest.Create("spike_recorder")
+E_recorder.set(recorder_params)
+
+I_recorder = nest.Create("spike_recorder")
+I_recorder.set(recorder_params)
+
+# Set initial membrane potentials (randomly distributed)
+print("Configuring neuron parameters...")
+all_neurons = E_neurons + I_neurons
+V_m_values = np.random.normal(mean_potential, sigma_potential, len(all_neurons))
+# Ensure values are below threshold
+V_m_values = np.clip(V_m_values, -np.inf, model_params["V_th"] - 0.1)
+all_neurons.V_m = V_m_values.tolist()
+
+# Calculate connection counts
+CE = int(NE * epsilon)  # excitatory connections per neuron
+CI = int(NI * epsilon)  # inhibitory connections per neuron
+
+# Create custom synapse models
+nest.CopyModel("static_synapse", "syn_ex", {"weight": JE})
+nest.CopyModel("static_synapse", "syn_in", {"weight": -JE * g})
+nest.SetDefaults("syn_ex", {"delay": delay})
+nest.SetDefaults("syn_in", {"delay": delay})
+
+# Set up STDP synapse
+nest.SetDefaults("stdp_synapse", stdp_params)
+
+# Determine synapse model for E->E connections
+if plastic_synapses:
+    synapse_model = "stdp_synapse"
+else:
+    synapse_model = "syn_ex"
+
+print("Connecting stimulus generators...")
+nest.Connect(E_stimulus, E_neurons, conn_spec="all_to_all", syn_spec="syn_ex")
+nest.Connect(I_stimulus, I_neurons, conn_spec="all_to_all", syn_spec="syn_ex")
+
+print("Connecting excitatory -> excitatory population...")
+# E->E with STDP and random initial weights
+weight_dist = nest.CreateParameter("normal", {"mean": JE, "std": sigma_w})
+nest.Connect(
+    E_neurons,
+    E_neurons,
+    conn_spec={"rule": "fixed_indegree", "indegree": CE},
+    syn_spec={"synapse_model": synapse_model, "weight": weight_dist},
+)
+
+print("Connecting excitatory -> inhibitory population...")
+nest.Connect(
+    E_neurons,
+    I_neurons,
+    conn_spec={"rule": "fixed_indegree", "indegree": CE},
+    syn_spec="syn_ex",
+)
+
+print("Connecting inhibitory -> excitatory population...")
+nest.Connect(
+    I_neurons,
+    E_neurons,
+    conn_spec={"rule": "fixed_indegree", "indegree": CI},
+    syn_spec="syn_in",
+)
+
+print("Connecting inhibitory -> inhibitory population...")
+nest.Connect(
+    I_neurons,
+    I_neurons,
+    conn_spec={"rule": "fixed_indegree", "indegree": CI},
+    syn_spec="syn_in",
+)
+
+print("Connecting spike recorders...")
+nest.Connect(E_neurons[:Nrec], E_recorder)
+nest.Connect(I_neurons[:Nrec], I_recorder)
+
+build_time = time.time()
+
+###############################################################################
+# Run simulation
+
+print("Simulating...")
+start_sim = time.time()
+nest.Simulate(simtime)
+sim_time = time.time() - start_sim
+build_time = time.time() - build_time
```

Comment:
Can't the build and simulation time not be directly read out from kernel status? Would save all these lines here and the import of the `time` module.